### PR TITLE
Fix/frontend baseplot

### PR DESCRIPTION
**Pull request by Tanguylo**

### CODE REVIEW

File: src/multiplots.ts [hunk 1]
```diff
@@ -1828,9 +1823,11 @@ export class MultiplePlots {
       // this.canvas.setAttribute('tabindex', '0');
       // this.canvas.focus();
 
-      for (let i=0; i<this.nbObjects; i++) {
-        this.objectList[i].mouse_interaction(this.objectList[i].isParallelPlot);
-      }
+      // for (let i = 0 ; i < this.nbObjects ; i++) {
```
Comment (GhislainJ) [inline src/multiplots.ts:1826]: Could remove these comments ?

File: src/subplots.ts [hunk 2]
```diff
@@ -1585,15 +1584,33 @@ export class BasePlot extends PlotData {
...
+    this.computeMovingObjects();
+
+    this.context_show.setTransform(this.movingMatrix);
+    this.drawMovingObjects();
+
+    this.context_show.setTransform(this.canvasMatrix);
     this.drawAxes();
+    
+    this.context_show.resetTransform();
+    // if (this.buttons_ON) { this.drawButtons(context) }
```
Comment (GhislainJ) [inline src/subplots.ts:1612]: Forgotten comment ?

File: src/subplots.ts [hunk 3]
```diff
@@ -1657,21 +1674,22 @@ export class BasePlot extends PlotData {
...
+          this.draw();
         }
-        this.draw();
         var is_inside_canvas = (e.offsetX >= this.X) && (e.offsetX <= this.width + this.X) && (e.offsetY >= this.Y) && (e.offsetY <= this.height + this.Y);
         if (!is_inside_canvas) {
           isDrawing = false;
-          this.axes.forEach(axis => {axis.saveLocation()});
-          this.translation = new Vertex(0, 0);
-          canvas.style.cursor = 'default';
+          // this.axes.forEach(axis => { axis.saveLocation() });
```
Comment (GhislainJ) [inline src/subplots.ts:1691]: Forgotten comment ?

File: src/subplots.ts [hunk 4]
```diff
@@ -1828,6 +1846,7 @@ export class Frame extends BasePlot {
     const [frameOrigin, xEnd, yEnd] = this.setFrameBounds();
     this.axes[0].transform(frameOrigin, xEnd);
     this.axes[1].transform(frameOrigin, yEnd);
+    console.log('done')
```
Comment (GhislainJ) [inline src/subplots.ts:1849]: Intempestive log ?

File: src/utils.ts [hunk 5]
```diff
@@ -2020,6 +2019,12 @@ export class newAxis {
     return Math.abs(this.origin.x - this.end.x);
   }
 
+  private get drawingColor(): string {
+    let color = this.strokeStyle;
+    if (this.mouseStyleON) { color = this.isHovered ? this.hoverStyle : this.isClicked ? this.clickedStyle : this.strokeStyle };
```
Comment (GhislainJ) [inline src/utils.ts:2024]: I think I've already seen this somewhere else in another PR. Can it be mutualized ?

File: src/plot-data.ts [hunk 6]
```diff
@@ -2110,11 +2109,10 @@ export abstract class PlotData {
         this.zoom_box_y = Math.min(mouse1Y, mouse2Y);
         this.zoom_box_w = Math.abs(mouse2X - mouse1X);
         this.zoom_box_h = Math.abs(mouse2Y - mouse1Y);
-        this.draw();
-
         canvas.style.cursor = 'crosshair';
         mouse_moving = true;
       }
+      // this.draw();
```
[the remark this answers is not in the document]
Comment (Tanguylo) [reply src/plot-data.ts:2100]: This is vowed to disappear. I think it is modified with pre-commit

File: src/subplots.ts (hunk 2 above)
Comment (Tanguylo) [reply src/subplots.ts:1616]: I tried to put buttons but stopped when i saw how it was done in current versions.
I let it here to remind to implement it and also where to write instructions. I can remove if required

File: src/subplots.ts (hunk 3 above)
Comment (Tanguylo) [reply src/subplots.ts:1697]: Removed

File: src/subplots.ts (hunk 4 above)
Comment (Tanguylo) [reply src/subplots.ts]: Removed

File: src/utils.ts (hunk 5 above)
Comment (Tanguylo) [reply src/utils.ts:2130]: Axis can be seen as an extent of Shape, which has already this kind of choice. So i see what you mean but i hope you will let me deal with that later, when the current plot_data will be less than 5000 rows

File: src/subplots.ts [hunk 7]
```diff
@@ -1649,33 +1671,30 @@ export class BasePlot extends PlotData {
...
-          this.axes.forEach(axis => {axis.saveLocation()});
-          this.translation = new Vertex(0, 0);
-          canvas.style.cursor = 'default';
-        }
+        const mouseInCanvas = (e.offsetX >= this.X) && (e.offsetX <= this.width + this.X) && (e.offsetY >= this.Y) && (e.offsetY <= this.height + this.Y);
+        if (!mouseInCanvas) { isDrawing = false };
       });
 
       canvas.addEventListener('mousedown', e => {
+        console.log(e.offsetX, e.offsetY);
```
Comment (GhislainJ) [inline src/subplots.ts:1697]: Leftover log ?

**Review by GhislainJ — APPROVED** [review]

File: src/subplots.ts (hunk 7 above)
Comment (Tanguylo) [reply src/subplots.ts:1697]: removed

File: src/multiplots.ts [hunk 8]
```diff
@@ -1379,18 +1372,18 @@ export class MultiplePlots {
...
-        let obj = this.objectList[i];
-        if (obj.type_ === 'scatterplot') {
-          MultiplotCom.frame_to_scatter_communication(frame, obj);
-        } else if (obj.type_ === 'parallelplot') {
-          MultiplotCom.frame_to_pp_communication(frame, obj);
-        } else if (obj.type_ === "frame") {
-          MultiplotCom.frame_to_frame_communication(frame, obj);
-        }else if (obj.type_ === "primitivegroupcontainer") {
-          primitive_indices.push(i);
+      this.objectList.forEach((plot, plotIndex) => {
```
[the remark this answers is not in the document]
Comment (Tanguylo) [reply src/multiplots.ts:1375]: Not yet... well it can but i think it is too early

**Review by GhislainJ — APPROVED** [review]

**Pull request merged**